vendor/k8s.io/kubernetes/cmd/kubelet/app/patch_volumequota.go

```diff
@@ -0,0 +1,95 @@
+package app
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"github.com/golang/glog"
+	yaml "gopkg.in/yaml.v2"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/kubernetes/pkg/volume"
+	"k8s.io/kubernetes/pkg/volume/emptydirquota"
+)
+
+// VolumeConfig contains options for configuring volumes on the node.
+type VolumeConfig struct {
+	metav1.TypeMeta
+
+	// LocalQuota contains options for controlling local volume quota on the node.
+	LocalQuota LocalQuota `yaml:"localQuota"`
+}
+
+// LocalQuota contains options for controlling local volume quota on the node.
+type LocalQuota struct {
+	// PerFSGroup can be specified to enable a quota on local storage use per unique FSGroup ID.
+	// At present this is only implemented for emptyDir volumes, and if the underlying
+	// volumeDirectory is on an XFS filesystem.
+	PerFSGroup *resource.Quantity `yaml:"perFSGroup"`
+}
+
+// THIS IS PART OF AN OPENSHIFT CARRY PATCH
+// PatchVolumePluginsForLocalQuota checks if the node config specifies a local storage
+// perFSGroup quota, and if so will test that the volumeDirectory is on a
+// filesystem suitable for quota enforcement. If checks pass the k8s emptyDir
+// volume plugin will be replaced with a wrapper version which adds quota
+// functionality.
+func PatchVolumePluginsForLocalQuota(rootdir string, plugins *[]volume.VolumePlugin) error {
+	volumeConfigFilePath := path.Join(rootdir, "volume-config.yaml")
+
+	if _, err := os.Stat(volumeConfigFilePath); os.IsNotExist(err) {
+		return nil
+	}
+
+	volumeConfigFile, err := ioutil.ReadFile(volumeConfigFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read %s: %v", volumeConfigFilePath, err)
+	}
+
+	var volumeConfig VolumeConfig
+	err = yaml.Unmarshal(volumeConfigFile, &volumeConfig)
```

Comment:
Make a group and version for this file and require them to be set as you expect.  This will allow you to version it later if need be.

Reply:
VolumeConfig in kubelet.config.openshift.io/v1 comes mind.